src/bin/pg_dump/pg_dump.c

```diff
@@ -16544,12 +16544,15 @@ dumpTableSchema(Archive *fout, const TableInfo *tbinfo)
 			 */
 			if (tbinfo->attoptions[j][0] != '\0')
 			{
-				fixAttoptionsBbfOriginalName(fout, &tbinfo->attoptions[j]);
+				char *attoptions = fixAttoptionsBbfOriginalName(fout, tbinfo->dobj.catId.oid, tbinfo->attnames[j]);
```

Comment:
it would less intrusive to use the old code where attoptions is overwritten with correct value. We can pass tbinfo and index j to babel function and let it take care of everything else.

Reply:
agree